data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/BinlogEventListener.java

```diff
@@ -130,12 +154,17 @@ void handleInsertEvent(com.github.shyiko.mysql.binlog.event.Event event) {
                     eventTimestampMillis,
                     eventTimestampMillis);
             addToBuffer(new Record<>(pipelineEvent));
+            eventCount++;
         }
+        bytesProcessedSummary.record(bytes);
```

Comment:
this should be moved after `flushBuffer(eventCount)` in case of any exceptions. I'm assuming flush buffer has infinite retry, and exceptions should not occur ?

Reply:
I checked dynamodb source [does it before flush](https://github.com/opensearch-project/data-prepper/blob/8f33225a32bc4ba226f6f83eb88d3ffff5212ad6/data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/StreamRecordConverter.java#L102), so trying to be consistent here.